fem/intrules_cut.hpp

```diff
@@ -116,6 +121,353 @@ class CutIntegrationRules
    virtual ~CutIntegrationRules() {}
 };
 
+#ifdef MFEM_USE_ALGOIM
+// define templated element bases
+namespace TmplPoly_1D
```

Comment:
Since this is ALGOIM specific, we could use a prefix to indicate that instead of `Tmpl`

Reply:
it can be used by other integration rules as well - it is not algoim specific. It is very useful class in case we have to use AD with in expressions with shape functions.